artemis/modules/sql_injection_detector.py

```diff
@@ -0,0 +1,187 @@
+
+    @staticmethod
+    def create_url_with_batch_payload(
+        url: str, param_batch: Optional[tuple[Any]], sleep_payload: Optional[str] = None
+    ) -> str:
+        if param_batch is not None:
+            assignments = {
+                key: sleep_payload if sleep_payload else random.choice(Config.Modules.SqlInjectionDetector.PAYLOADS)
+                for key in param_batch
+            }
+            url_with_payload = f"{url}?" + "&".join([f"{key}={value}" for key, value in assignments.items()])
+            return unquote(url_with_payload)
+        return url
+
+    @staticmethod
+    def is_url_with_payload(url: str) -> bool:
+        if re.search("/?/*=", url):
+            return True
+        return False
+
+    @staticmethod
+    def change_url_params(url: str, sleep_payload: Optional[str] = None) -> str:
+        parsed_url = urlparse(url)
+        query_params = parse_qs(parsed_url.query)
+        params = list(query_params.keys())
+        new_query_params = {}
+
+        for param in params:
+            new_query_params[param] = [
+                sleep_payload if sleep_payload else random.choice(Config.Modules.SqlInjectionDetector.PAYLOADS)
+            ]
+
+        new_query_string = urlencode(new_query_params, doseq=True)
+        new_url = urlunparse(
+            (
+                parsed_url.scheme,
+                parsed_url.netloc,
+                parsed_url.path,
+                parsed_url.params,
+                new_query_string,
+                parsed_url.fragment,
+            )
+        )
+        return unquote(new_url)
+
+    @staticmethod
+    def is_response_time_within_threshold(elapsed_time: float) -> bool:
+        if elapsed_time < Config.Modules.SqlInjectionDetector.THRESHOLD:
+            return True
+        return False
+
+    def are_requests_time_efficient(self, url: str) -> bool:
+        start = timer()
+        self.response_when_injecting(url)
+        elapsed_time = datetime.timedelta(seconds=timer() - start).seconds
+
+        flag = self.is_response_time_within_threshold(elapsed_time)
+
+        return flag
+
+    @staticmethod
+    def contains_error(response: HTTPResponse) -> bool | None:
+        for message in SQL_ERROR_MESSAGES:
+            if re.search(message, response.content):
+                return True
+        return False
+
+    @staticmethod
+    def response_when_injecting(url: str) -> HTTPResponse:
+        response = http_requests.get(url)
+
+        return response
+
+    def scan(self, urls: List[str], task: Task) -> Dict[str, object]:
+        task_result = {
+            "task_host": get_target_url(task),
+            "status": task.status,
+            "message": [],
+        }
+        message = []
+        for current_url in urls:
+            #  Below is the code describing the logic, but it is repetitive. What is better (create a universal
+            #  function or leave it as below)
```

Comment:
it's fine, just write a comment why the top if

Reply:
Done.